apps/pace/app.ts

```diff
@@ -0,0 +1,226 @@
+{
+const Layout = require("Layout");
+const time_utils = require("time_utils");
+const exs = require("exstats").getStats(
+  ["dist", "pacec"],
+  {
+   notify: {
+      dist: {
+        increment: 1000,
+      },
+    },
+  },
+);
+const S = require("Storage");
+
+let drawTimeout: TimeoutId | undefined;
+
+type Dist = number & { brand: 'dist' };
+type Time = number & { brand: 'time' };
+
+type Split = {
+  dist: Dist,
+  time: Time,
+};
+
+const splits: Split[] = []; // times
+let splitOffset = 0, splitOffsetPx = 0;
+
+const GPS_TIMEOUT_MS = 30000;
+
+const layout = new Layout({
+  type: "v",
+  c: [
+    {
+      type: "txt",
+      font: "6x8:2",
+      label: "Pace",
+      id: "paceLabel",
+      pad: 4
+    },
+    {
+      type: "txt",
+      font: "Vector:40",
+      label: "",
+      id: "pace",
+      halign: 0
+    },
+    {
+      type: "txt",
+      font: "6x8:2",
+      label: "Time",
+      id: "timeLabel",
+      pad: 4
+    },
+    {
+      type: "txt",
+      font: "Vector:40",
+      label: "",
+      id: "time",
+      halign: 0
+    },
+  ]
+}, {
+  lazy: true
+});
+
+const draw = () => {
+  if (!exs.state.active) {
+    // no draw-timeout here, only on user interaction
+    drawSplits();
+    return;
+  }
+
+  if (drawTimeout) clearTimeout(drawTimeout);
+  drawTimeout = setTimeout(draw, 1000);
+
+  const now = Date.now();
+
+  let pace: string;
+  if ("time" in exs.state.thisGPS
+  && now - (exs.state.thisGPS.time as unknown as number) < GPS_TIMEOUT_MS)
+  {
+    pace = exs.stats.pacec.getString()
+  }else{
+    pace = "No GPS";
+  }
+
+  layout["time"]!.label = formatDuration(exs.state.duration);
+  layout["pace"]!.label = pace;
+  g.clearRect(Bangle.appRect); // hack for broken layout rendering of `pace`
```

Comment:
I think this is what makes all but the bottom stopwatch clear and not reappear when I have no GPS signal.

I'd expect the labels and "No GPS" message to keep showing.

Reply:
Ah yes - thought I'd dropped that commit, gone now. Thanks for the spot!